runtime/services/catalog/migrator/metricsviews/metrics_views.go

```diff
@@ -62,13 +59,16 @@ func (m *metricsViewMigrator) Validate(ctx context.Context, olap drivers.OLAPSto
 		return migrator.CreateValidationError(catalog.Path, err.Error())
 	}
 
-	fieldsMap := make(map[string]*runtimev1.StructType_Field)
-	for _, field := range model.Schema.Fields {
-		fieldsMap[field.Name] = field
-	}
+	// if a time dimension is selected it should exist
+	if mv.TimeDimension != "" {
+		fieldsMap := make(map[string]*runtimev1.StructType_Field)
+		for _, field := range model.Schema.Fields {
+			fieldsMap[field.Name] = field
+		}
 
-	if _, ok := fieldsMap[mv.TimeDimension]; !ok {
-		return migrator.CreateValidationError(catalog.Path, TimestampNotFound)
+		if _, ok := fieldsMap[mv.TimeDimension]; !ok {
+			return migrator.CreateValidationError(catalog.Path, TimestampNotFound)
+		}
```

Comment:
If `fieldsMap` is only used for this check, it would be more performant to just do:

```go
found := false
for _, field := range model.Schema.Fields {
  if field == mv.TimeDimension { // NOTE: replace with normalized compare
    found = true
    break
  }
}

if !found {
  return migrator.CreateValidationError(catalog.Path, TimestampNotFound)
}
```

Reply:
huh. i probably had dimension validation use the same map. perhaps we should do that to improve the check there. will be pretty bad when dashboard is created from a large data set.